internal/provider/config_data_source_test.go

```diff
@@ -471,8 +471,8 @@ func TestUnify(t *testing.T) {
 			"foo=1.2.3",
 		},
 		wantDiag: []diag.Diagnostic{
-			diag.NewWarningDiagnostic("unable to lock certain packages for amd64", "[intel-fast-as-f-math]"),
-			diag.NewWarningDiagnostic("unable to lock certain packages for arm64", "[arm-energy-efficient-as-f-arithmetic]"),
+			diag.NewErrorDiagnostic("unable to lock certain packages for amd64", "[intel-fast-as-f-math]"),
+			diag.NewErrorDiagnostic("unable to lock certain packages for arm64", "[arm-energy-efficient-as-f-arithmetic]"),
```

Comment:
These were intended as examples where we legitimately shouldn't fail because it is actually impossible to create a platform-independent lock file.